concordium-smart-contract-testing/src/types.rs

```diff
@@ -292,8 +292,11 @@ impl ContractInvokeSuccess {
     /// Extract all the events logged by all the contracts in the invocation.
     /// The events are returned in the order that they are emitted, and are
     /// paired with the address of the contract that emitted it.
+    ///
+    /// Only events from successful trace elements are included. See
+    /// [`Self::success_trace_elements`] for more details.
     pub fn events(&self) -> impl Iterator<Item = (ContractAddress, &[ContractEvent])> {
```

Comment:
Might be useful to also have a helper to check if any rollback occurred/any inner failures occurred.